src/main/java/com/dawsonsystems/session/MongoManager.java

```diff
@@ -552,13 +577,18 @@ private void initDbConnection(String path) throws LifecycleException {
               .retryReads(false)
               .retryWrites(false);
 
-      List<MongoCredential> mongoCredentials = new ArrayList<>();
-
-      if (sslTrustStorePem != null && sslTrustStorePem.length() > 0 && sslKeyStorePem != null && sslKeyStorePem.length() > 0) {
+      Optional<SSLContext> sslContext = null;
+      if (Boolean.parseBoolean(useJndiSslContext)) {
+        sslContext = fetchMongoSSLContextFromJndi();
+      } else if (sslTrustStorePem != null && sslTrustStorePem.length() > 0 && sslKeyStorePem != null && sslKeyStorePem.length() > 0) {
         log.info("Detected Mongo SSL configuration: trust store PEM: " + sslTrustStorePem + ", key store PEM: " + sslKeyStorePem);
-        SSLContext sslContext = sslContextFromPEMs(new File(sslTrustStorePem), new File(sslKeyStorePem));
+        sslContext = Optional.of(sslContextFromPEMs(new File(sslTrustStorePem), new File(sslKeyStorePem)));
+      }
+
+      List<MongoCredential> mongoCredentials = new ArrayList<>();
+      if (sslContext != null && sslContext.isPresent()) {
```

Comment:
I think you can remove this null check, since both code paths return an Optional

Reply:
I think it handles scenario where Optional object itself is null

Reply:
Yeah, but the main point of using Optional is to avoid null checks. You should generally do one or the other